LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DefaultLoRaDataRequestHandler.cs

```diff
@@ -789,7 +883,9 @@ private async Task<bool> DetermineIfFramecounterIsFromNewlyStartedDeviceAsync(Lo
                         // known problem when device restarts, starts fcnt from zero
                         // We need to await this reset to avoid races on the server with deduplication and
                         // fcnt down calculations
-                        _ = await frameCounterStrategy.ResetAsync(loRaDevice, payloadFcnt, this.configuration.GatewayID);
+                        if (concentratorDeduplicationResult is ConcentratorDeduplicationResult.NotDuplicate)
+                            _ = await frameCounterStrategy.ResetAsync(loRaDevice, payloadFcnt, this.configuration.GatewayID);
+
                         isFrameCounterFromNewlyStartedDevice = true;
```

Comment:
Shouldn't this also be be in the previous if clause (concentrator?)

Reply:
It was not a programming error and there are 3 tests for unconfirmed first messages that fail if we add this in the previous if. However there might be a bug there: can we ever get a resubmission for an unconfirmed message? Only if it's a restarted device. But even then it would not send the exact same message, right?

Reply:
Discussed this with Patrick and he said that resubmission of unconfirmed message with frame counter 1 (first message) is ok to pass through a second time since we are using `IsABPRelaxedFrameCounter`

Reply:
Added an explanation and curly braces to the if to make it more clear (fb4534a743c117c2f4f1216e939f6672c2577651)